class.jetpack-post-images.php

```diff
@@ -339,21 +345,49 @@ static function from_html( $html_or_id ) {
 			return $images;
 		}
 
-		preg_match_all( '!<img.*src=[\'"]([^"]+)[\'"].*/?>!iUs', $html, $matches );
-		if ( !empty( $matches[1] ) ) {
-			foreach ( $matches[1] as $match ) {
-				if ( stristr( $match, '/smilies/' ) )
-					continue;
+		// Do not go any further if DOMDocument is disabled on the server.
+		if ( ! class_exists( 'DOMDocument' ) ) {
+			return $images;
+		}
 
-				$images[] = array(
-					'type'  => 'image',
-					'from'  => 'html',
-					'src'   => html_entity_decode( $match ),
-					'href'  => '', // No link to apply to these. Might potentially parse for that as well, but not for now
-				);
+		// Let's grab all image tags from the HTML.
+		$dom_doc = new DOMDocument;
+		$dom_doc->loadHTML( $html );
+		$image_tags = $dom_doc->getElementsByTagName( 'img' );
+
+		// For each image Tag, make sure it can be added to the $images array, and add it.
+		foreach ( $image_tags as $image_tag ) {
+			$img_src = $image_tag->getAttribute( 'src' );
+
+			if ( empty( $img_src ) ) {
+				continue;
 			}
-		}
 
+			// Do not grab smiley images that were automatically created by WP when entering text smilies.
+			if ( stripos( $img_src, '/smilies/' ) ) {
+				continue;
+			}
+
+			$meta['width']  = (int) $image_tag->getAttribute( 'width' );
```

Comment:
I can't see `$meta` defined previously, what do you say we `$meta = array(` here?

Reply:
Good call. Done in 456c026e62be36078e42f502eef0cb9e60b42332